trinity/protocol/common/managers.py

```diff
@@ -64,6 +70,23 @@ def add(self, time: float, size: int) -> None:
         self.total_response_time += time
 
 
+@asynccontextmanager
+async def disconnect_on_misbehavior(
+        peer: BasePeer,
+        command_class: Type[Command],
+        logger: logging.Logger) -> AsyncGenerator[None, None]:
+    try:
+        yield
+    except MalformedMessage as err:
```

Comment:
I was surprised to see a MalformedMessage being handled at this level, as I thought that was usually handled at a lower level, closer to where it's actually raised (the decoding of messages on `BasePeer.read_msg()`), and then I found out the manager's `_normalize_response()` methods raise that as well, but that doesn't sound right as that exception should be used to signal a msg we can't decode (`Raised when a p2p command is received with a malformed message`). How would you feel about using a separate exception to indicate an unexpected response at the managers level, @pipermerriam?

Reply:
:+1: the use of `MalformedMessage` here isn't appropriate, it was just convenient and felt better than `ValidationError` at the time.

Also worth pointing out.  This exception handling should be extremely unlikely and in the event that it is encountered, it is more likely to be a problem with our code than a problem with the peer since the decoded messages have a fixed and known format since they come directly from the decoded command message.  

IE, if we expected a tuple of block headers and we find we have some other type, something is probably wrong in our code that is allowing us to pass the wrong value into this API.

Reply:
Yeah, on further inspection, this check appears entirely redundant. The `Command` enforces the correct type when decoding the RLP. I think we can reasonably get rid of the `MalformedMessage` part of the normalization step, or at least drop the bit that disconnects from the peer.